Model/Config/Backend/PaymentMethodsStatus.php

```diff
@@ -27,50 +23,25 @@
 class PaymentMethodsStatus extends Value
 {
     protected PaymentMethods $paymentMethodsHelper;
-    private WriterInterface $configWriter;
-
-    /*
-     * Following payment methods should be enabled with their own configuration path.
-     */
-    const EXCLUDED_PAYMENT_METHODS = [
-        AdyenPayByLinkConfigProvider::CODE,
-        AdyenPosCloudConfigProvider::CODE,
-        AdyenMotoConfigProvider::CODE
-    ];
 
     public function __construct(
         Context $context,
         Registry $registry,
         ScopeConfigInterface $config,
         TypeListInterface $cacheTypeList,
-        WriterInterface $configWriter,
         PaymentMethods $paymentMethodsHelper,
         AbstractResource $resource = null,
         AbstractDb $resourceCollection = null,
         array $data = []
     ) {
-        $this->configWriter = $configWriter;
         $this->paymentMethodsHelper = $paymentMethodsHelper;
 
         parent::__construct($context, $registry, $config, $cacheTypeList, $resource, $resourceCollection, $data);
     }
 
     public function afterSave(): PaymentMethodsStatus
     {
-        $value = $this->getValue();
-        $adyenPaymentMethods = $this->paymentMethodsHelper->getAdyenPaymentMethods();
-
-        foreach ($adyenPaymentMethods as $adyenPaymentMethod) {
-            // Exclude following payment methods. They need to use their own configuration path.
-            if (in_array($adyenPaymentMethod, self::EXCLUDED_PAYMENT_METHODS)) {
-                continue;
-            }
-
-            $this->configWriter->save(
-                'payment/' . $adyenPaymentMethod . '/active',
-                $value,
-            );
-        }
+        $this->paymentMethodsHelper->togglePaymentMethodsActivation($this->getValue());
```

Comment:
`togglePaymentMethodsActivation()` expects a boolean and I think `$this->getValue()` is a string with value either `'1'` or `'0'`